neard/src/runtime.rs

```diff
@@ -523,6 +532,30 @@ impl NightshadeRuntime {
 
         Ok(result)
     }
+
+    #[allow(unused_variables)]
+    fn check_duplicate_transaction(
+        &self,
+        state_update: &TrieUpdate,
+        tx: &SignedTransaction,
+    ) -> Result<bool, Error> {
+        #[cfg(feature = "protocol_feature_transaction_hashes_in_state")]
+        {
+            get::<()>(
+                &state_update,
+                &TrieKey::AccountTransactionHash {
+                    account_id: tx.transaction.signer_id.to_string(),
+                    hash: tx.get_hash(),
+                },
+            )
+            .map(|x| x.is_some())
+            .map_err(|e| Error::from(ErrorKind::StorageError(e)))
```

Comment:
This feels like abuse of store with assumption that `()` is encoded as nothing in borsh. I believe it would be cleaner if store would expose `contains` method just to check the existance of a given key.

Notes:

1. `keyMayExist` in RocksDB is [confusing](https://github.com/facebook/rocksdb/blob/v5.14.2/include/rocksdb/db.h#L369-L375), so we definitely don't want to use it (see https://github.com/facebook/rocksdb/issues/4149)
2. I think we can use [`get_pinned`](https://docs.rs/rocksdb/0.15.0/rocksdb/struct.DB.html?search=#method.get_pinned) method to avoid memory copies (and borsh deserialization) since we don't even want to read the value itself

Reply:
> This feels like abuse of store with assumption that () is encoded as nothing in borsh

I don't think I assume that. Can you explain why?

Reply:
My bad, sorry. I thought that you store something else but use `()` in `get` just to "skip" heavy deserialization